djangocms_versioning/admin.py

```diff
@@ -737,14 +736,17 @@ def discard_view(self, request, object_id):
             return render(
                 request, "djangocms_versioning/admin/discard_confirmation.html", context
             )
-        else:
-            version_url = version_list_url(version.content)
-            if request.POST.get("discard"):
-                version.content.delete()
-                version.delete()
 
-            # Redirect
-            return redirect(version_url)
+        version_url = version_list_url(version.content)
+        if request.POST.get("discard"):
+            ModelClass = version.content.__class__
+            deleted = version.delete()
+            if deleted[1]['last']:
+                # need a success message as well.
```

Comment:
```suggestion

```

Reply:
This should be deleted.

Reply:
@Aiky30 You can commit this change. Apparently only users with write permission can commit things from github.

Reply:
I also cannot commit this suggestion because I don't have the permission on your repo.